src/common/function/test/FunctionManagerTest.cpp

```diff
@@ -818,657 +821,175 @@ TEST_F(FunctionManagerTest, time) {
-  }
-  {
-    auto result = FunctionManager::getReturnType("endNode", {Value::Type::NULLVALUE});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::NULLVALUE, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("keys", {Value::Type::VERTEX});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("keys", {Value::Type::EDGE});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("keys", {Value::Type::MAP});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("nodes", {Value::Type::PATH});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("reverse", {Value::Type::LIST});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("tail", {Value::Type::LIST});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("relationships", {Value::Type::PATH});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("head", {Value::Type::LIST});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::__EMPTY__, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("last", {Value::Type::LIST});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::__EMPTY__, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("coalesce", {Value::Type::LIST});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::__EMPTY__, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType("range", {Value::Type::INT, Value::Type::INT});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
-  }
-  {
-    auto result = FunctionManager::getReturnType(
-        "range", {Value::Type::INT, Value::Type::INT, Value::Type::INT});
-    ASSERT_TRUE(result.ok()) << result.status();
-    EXPECT_EQ(Value::Type::LIST, result.value());
+    struct TestReurnType {
+      std::string func_name;          // function being tested
+      std::vector<Value::Type> args;  // parameters for test
+      Value::Type returnType;         // the correct return type expected
+    };
+
+    // add test case
+    std::vector<TestReurnType> test_cases = {
+        TestReurnType{"abs", {Value::Type::INT}, Value::Type::INT},
```

Comment:
It would be much more helpful if TestReturnType run the test directly. The `TestReturnType` could be a macro or util function. And we need to easily know which one case failed, that means something like line numbers.

Reply:
It can useful that adding info about func for locating